### PR TITLE
Ability to inject the authorization URL through the constructor

### PR DESCRIPTION
**Pull request by jefersonralmeida**
I've just added the $urlAuthorize property as a optional constructor parameter.

It fixes this [issue](https://github.com/greew/oauth2-azure-provider/issues/2).

### CODE REVIEW

File: src/Provider/Azure.php [hunk 1]
```diff
@@ -28,11 +28,22 @@ class Azure extends AbstractProvider
...
 
     /** @var string */
     protected $urlResourceOwnerDetails = 'https://graph.microsoft.com/v1.0/me';
 
+    /**
+     * @param array $options
+     * @param array $collaborators
+     * @param string $urlAuthorize Base url for authorization.
+     */
+    public function __construct(array $options = [], array $collaborators = [], string $urlAuthorize = 'https://login.microsoftonline.com')
```
Comment (jefersonralmeida) [inline src/Provider/Azure.php]: Just letting clear that this syntax could be much cleaner, using constructor property promotion, but I saw that the package composer file requires support to PHP 7 (property promotion is only supported by php 8.x).

**Comment by greew** [comment]
Great - thanks! :smiley: 

Just need to break the constructor properties up into single lines to conform to the PHP CodeSniffer rules.

When that's fixed, I'll merge it!

**Comment by jefersonralmeida** [comment]
It's done @greew , sorry for the delay, I was out on a trip last week.

**Comment by jefersonralmeida** [comment]
Hey @greew , can we merge it?

**Comment by greew** [comment]
Yeah - Thank you for your work! 😊

**Pull request merged**